crates/uv-publish/src/lib.rs

```diff
@@ -247,46 +258,57 @@ pub async fn check_trusted_publishing(
     trusted_publishing: TrustedPublishing,
     registry: &Url,
     client: &BaseClient,
-) -> Result<Option<TrustedPublishingToken>, PublishError> {
+) -> Result<TrustedPublishResult, PublishError> {
     match trusted_publishing {
         TrustedPublishing::Automatic => {
             // If the user provided credentials, use those.
             if username.is_some()
                 || password.is_some()
                 || keyring_provider != KeyringProviderType::Disabled
             {
-                return Ok(None);
+                return Ok(TrustedPublishResult::Skipped);
             }
             // If we aren't in GitHub Actions, we can't use trusted publishing.
             if env::var(EnvVars::GITHUB_ACTIONS) != Ok("true".to_string()) {
-                return Ok(None);
+                return Ok(TrustedPublishResult::Skipped);
             }
             // We could check for credentials from the keyring or netrc the auth middleware first, but
             // given that we are in GitHub Actions we check for trusted publishing first.
             debug!("Running on GitHub Actions without explicit credentials, checking for trusted publishing");
             match trusted_publishing::get_token(registry, client.for_host(registry)).await {
-                Ok(token) => Ok(Some(token)),
+                Ok(token) => Ok(TrustedPublishResult::Configured(token)),
                 Err(err) => {
                     // TODO(konsti): It would be useful if we could differentiate between actual errors
                     // such as connection errors and warn for them while ignoring errors from trusted
                     // publishing not being configured.
                     debug!("Could not obtain trusted publishing credentials, skipping: {err}");
-                    Ok(None)
+                    Ok(TrustedPublishResult::Ignored(err))
                 }
             }
         }
         TrustedPublishing::Always => {
             debug!("Using trusted publishing for GitHub Actions");
+
+            if username.is_some() {
+                return Err(PublishError::MixedCredentials("a username"));
+            }
+            if password.is_some() {
+                return Err(PublishError::MixedCredentials("a password"));
+            }
+            if keyring_provider != KeyringProviderType::Disabled {
+                return Err(PublishError::MixedCredentials("the keyring"));
+            }
```

Comment:
Would it make sense to try and combine these into one error? Otherwise, users might hit the first one, then fix it. Then hit the second one, then fix it. And so on.